File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        
        
        while (true) {
            int copyArr[] = arr.clone();
            
            for (int i = 0; i < arr.length; i++) {
                if (arr[i] >= 50 && arr[i] % 2 == 0) {
                    arr[i] /= 2;
                } else if (arr[i] < 50 && arr[i] % 2 == 1) {
                    arr[i] *= 2;
                    arr[i] += 1;
                }
            }
            
            if (Arrays.equals(copyArr, arr)) break;
            answer++;
        }
        
        
        return answer;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    var answer = 0;
    
    while(true) {
        let copyArr = arr.slice();
        
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] >= 50 && arr[i] % 2 === 0) {
                    arr[i] /= 2;
            } else if (arr[i] < 50 && arr[i] % 2 === 1) {
                arr[i] *= 2;
                arr[i] += 1;
            }
        }
        if (JSON.stringify(copyArr) === JSON.stringify(arr)) break;
        answer++;
    }
    
    return answer;
}